api/src/test/java/org/openmrs/api/PersonServiceTest.java

```diff
@@ -530,7 +530,7 @@ public void getSimilarPeople_shouldMatchN2InTwoNamesSearch() throws Exception {
 		updateSearchIndex();
 		
 		Set<Person> matches = Context.getPersonService().getSimilarPeople("D Graham", 1979, "M");
-		assertEquals(3, matches.size());
```

Comment:
Did you intentionally comment this out?

Reply:
@dkayiwa yes, I did so that the test case executed line 534 and you can more easily evaluate if it is as described here: 
https://talk.openmrs.org/t/soundex-search-in-lucenequeries/26068/45?u=fruether

In case we decide that the new logic is fine the 3 of the assertEqual part hast to be increased to 8.

Reply:
However, this state should be fine to do the checks before I go back on working on it.

Reply:
@dkayiwa any updated regarding this?
Shall I clarify more?

Reply:
@dkayiwa any roadmap when you will have a look to it as discussed here: https://talk.openmrs.org/t/soundex-search-in-lucenequeries/26068/45?u=fruether

Reply:
@fruether very sorry for the delay. Looking at it now. Do you still have time to take it forward?

Reply:
@dkayiwa sure thing. The implementation part will not be so heavy (it is almost done) it is more a decision of how to tune the parameters[1] and we can tolerate a less strict search algorithm that returns more rather then less values but in an order


[1] https://github.com/openmrs/openmrs-core/pull/3663/commits/6c3ae1915ef6a5347e63298aadb3917bc7513cb6#diff-5dda215a54adb5246394297fb22fcb960b56e2e160f7f11b611f8dce295935acR39

Reply:
@fruether i have closely looked at the extra results returned, and they look correct. So i believe that we can tolerate the less strict search algorithm that returns more rather then less values, but in an order.